### DISCUSSION TITLE
pkl.file

### DISCUSSION THREAD

**Discussion by ThomasGiavasopoulos** [do-mpc/do-mpc#250] [Q&A]
Good afternoon, 

I am designing the landing trajectory for a rocket. 

The results are stored and loaded in a pkl file as mentioned in the documentation:
`save_results([mpc, simulator])
!ls ./results/
results = load_results('./results/results.pkl')`

Is there a way to choose how the data is stored and is there an option to allow overwriting?

Also, when I try to take a look at the data, I am met with the following text (I am running the code in a jupyter notebook):
`<img width="901" alt="image" src="https://user-images.githubusercontent.com/57414585/159958023-0cbc0027-16ff-4b5a-bd2d-d4c2b08a51cd.png">`

Could you please redirect me to sources that will enlighten me on the topic of pkl files and how these can be "viewed" and their data "manipulated"?

Thank you very much!

**Comment by zmdntnu** [comment]
Dear @4flixt,
I also have a general question regarding saving data. The figure showed in the simulation using mpc_graphics.plot_results() is different from the one obtained using the results['mpc']['_x'], and the difference lies on the x axis. In the former plotting system, the x axis uses time as the unit while in the latter one, it's time step instead. Therefore, I wonder, how can I save the exact data as mpc_graphics.plot_results in time instead of time step. In my case, my mpc and simulator time step are both 5s, and it took 184 steps to finish the simulation, which gives me only 184 data in one column. But I need the original 184*5 data as showed in the mpc_graphics.plot_results. I hope you could understand my question, if not, please let me know. Hopefully, you could provide me a solution to this. 

Thank you very much.
